### ISSUE TITLE
Question about training difficulties for pose network

### ISSUE THREAD

**Issue by marcel1991** [tinghuiz/SfMLearner#39]
I have done my own implementation in torch of a similar model as yours that does train the depth and the camera pose transformation in the same unsupervised manner as you do. 

However I have difficulties with the pose network that is somehow not able to learn the transformation properly. My implementation is right, I tested it with stereo examples and it is able to learn the depth and the static transformation of the stereo setup. However if I go on to a not static setup, like a monocular video, the pose network is not able to learn the different transformation parameters anymore. That even happens if I use stereo images and change the left with right image randomly, so that it has to learn just 2 different transformations.

So I am asking you if you experienced any such difficulties with your pose network? Did you use the same learning rate for the depth and the pose network? And how did you normalize the image inputs (range between -1 and 1?)?

**Comment by tinghuiz** [comment]
How do the warped/projected images look in the non-static case? Also, did you visualize the histogram of your pose prediction values (tx, ty, tz, rx, ry, rz)? Do they look abnormal?

**Comment by marcel1991** [comment]
In the non-static case the depth predictions are infinity(or highest possible value) or zero. You can see that the depth prediction tries to somehow lower the reconstruction loss with these extreme values because the predicted transformation is wrong. 

I actually realized that the transformation becomes the same for every input. Looks like it's the mean of all possible transformations because it's mostly tz that is non-zero (tz is the forward moving camera which is the case most of the time in the training set). 
So over a batch the pose net gives always the same transformation predictions like it is not able to learn from the different inputs and it can just learn "the best fit" for all images in a batch together. 

So I thought the input value range is maybe not good (I use -0.5 to 0.5) or the learning rate for the pose net is too high.

**Comment by tinghuiz** [comment]
You can try scaling the predicted pose values to be even smaller (e.g. use 0.0001 in https://github.com/tinghuiz/SfMLearner/blob/master/nets.py#L45), and see if the depth prediction is still outrageously large.
